test/catalog/constraints_test.cpp

```diff
@@ -55,54 +75,848 @@ TEST_F(ConstraintsTests, NOTNULLTest) {
+
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+
+  // begin this transaction
+  auto txn = txn_manager.BeginTransaction();
+
+  // Test1: insert a tuple with column  meet the constraint requirment
+  bool hasException = false;
+  try {
+    ConstraintsTestsUtil::ExecuteOneInsert(
+        txn, data_table.get(), type::ValueFactory::GetIntegerValue(10));
+  } catch (ConstraintException e) {
+    hasException = true;
+  }
+  EXPECT_FALSE(hasException);
+
+  // Test2: insert not a valid column violate the constraint
+  hasException = false;
+  try {
+    ConstraintsTestsUtil::ExecuteOneInsert(
+        txn, data_table.get(), type::ValueFactory::GetIntegerValue(-1));
+  } catch (ConstraintException e) {
+    hasException = true;
+  }
+  EXPECT_TRUE(hasException);
+
+  // commit this transaction
+  txn_manager.CommitTransaction(txn);
+  delete data_table.release();
+}
+#endif
+
+#ifdef UNIQUE_TEST
+TEST_F(ConstraintsTests, UNIQUETest) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto catalog = catalog::Catalog::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog->CreateDatabase(DEFAULT_DB_NAME, nullptr);
+  auto column1 = catalog::Column(type::Type::INTEGER, 25, "A", false, 0);
+  auto column2 = catalog::Column(type::Type::INTEGER, 25, "B", false, 1);
+
+  auto constraints = catalog::Constraint(ConstraintType::UNIQUE, "unique1");
+  column1.AddConstraint(constraints);
+  std::cout << "****" << constraints.GetInfo() << std::endl;
+  std::unique_ptr<catalog::Schema> table_schema(
+      new catalog::Schema({column1, column2}));
+  std::string table_name("TEST_TABLE");
+  catalog::Catalog::GetInstance()->CreateTable(DEFAULT_DB_NAME, table_name,
+                                               std::move(table_schema), txn);
+  txn_manager.CommitTransaction(txn);
+  storage::Database *database = catalog->GetDatabaseWithName(DEFAULT_DB_NAME);
+  storage::DataTable *table = database->GetTableWithName(table_name);
+
+  // table->AddUNIQUEIndex();
+
+  txn = txn_manager.BeginTransaction();
+  // begin this transaction
+  // Test1: insert a tuple with column  meet the unique requirment
+  bool hasException = false;
+  try {
+    // bool result = true;
+    // result =
+    ConstraintsTestsUtil::ExecuteOneInsert(
+        txn, table, type::ValueFactory::GetIntegerValue(10));
+    // if (result == false) hasException = true;
+  } catch (ConstraintException e) {
+    hasException = true;
+  }
+  EXPECT_FALSE(hasException);
+
+  // Test2: insert not a valid column violate the constraint
+  hasException = false;
+  try {
+    // bool result = true;
+    // result =
+    ConstraintsTestsUtil::ExecuteOneInsert(
+        txn, table, type::ValueFactory::GetIntegerValue(10));
+    // if (result == false) hasException = true;
+  } catch (ConstraintException e) {
+    hasException = true;
+  }
+  EXPECT_TRUE(hasException);
+
+  hasException = false;
+  try {
```

Comment:
Other tests can be added for unique which check that when someone tries to update the values of a column to another value which already exists, your constraint checker should throw an exception.

Also, another test which :
- Inserts a value 10 and succeeds
- Tries to insert 10 again and fails
- the row with 10 in the column is deleted
- Once again an insert of 10 is done and it should succeed